llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -1761,6 +1761,11 @@ bool SimplifyCFGOpt::hoistSuccIdenticalTerminatorToSwitchOrIf(
       OtherSuccTI->replaceAllUsesWith(NT);
     NT->takeName(I1);
   }
+  // Also clone DPValues from the existing terminator, and all others (to
+  // duplicate existing hoisting behaviour).
+  NT->cloneDebugInfoFrom(I1);
+  for (Instruction *OtherSuccTI : OtherSuccTIs)
+    NT->cloneDebugInfoFrom(OtherSuccTI);
```

Comment:
Which test (assuming tests will be added shortly) covers this change?

Reply:
This should probably be split out of here - I think this is a general DPValue bug that I found while validating the assignment tracking patches; I'll create a separate patch with a reduced reproducer as the test case.

Reply:
Nevermind, I was wrong - `DebugInfo/Generic/assignment-tracking/simplifycfg/empty-block.ll` fails without this! Will re-add this to the patch (and also make sure that test is included).